example/src/main/java/com/ryanharter/auto/value/gson/example/PersonTypeAdapterFactory.java

```diff
@@ -0,0 +1,18 @@
+package com.ryanharter.auto.value.gson.example;
+
+import com.google.gson.Gson;
+import com.google.gson.TypeAdapter;
+import com.google.gson.TypeAdapterFactory;
+import com.google.gson.reflect.TypeToken;
+
+public class PersonTypeAdapterFactory implements TypeAdapterFactory {
```

Comment:
Call registerTypeAdapter on GsonBuilder

On Tue, Apr 19, 2016 at 9:41 PM Piasy <notifications@github.com> wrote:

> In
> example/src/main/java/com/ryanharter/auto/value/gson/example/PersonTypeAdapterFactory.java
> <https://github.com/rharter/auto-value-gson/pull/30#discussion_r60339958>:
>
> > @@ -0,0 +1,18 @@
> > +package com.ryanharter.auto.value.gson.example;
> > +
> > +import com.google.gson.Gson;
> > +import com.google.gson.TypeAdapter;
> > +import com.google.gson.TypeAdapterFactory;
> > +import com.google.gson.reflect.TypeToken;
> > +
> > +public class PersonTypeAdapterFactory implements TypeAdapterFactory {
>
> Sorry I just return to my laptop. How could we register a type adapter
> without factory? The generated GsonTypeAdapter need a gson parameter for
> its constructor, but before we build gson, we have no gson instance.
>
> —
> You are receiving this because you commented.
>
>
> Reply to this email directly or view it on GitHub
> <https://github.com/rharter/auto-value-gson/pull/30/files/e2d8ec9e7411524859dd3b2cfe98a76e6f3ce9e5#r60339958>
>